gateway/gateway.go

```diff
@@ -191,19 +190,11 @@ func NewGateway(
 		started: false,
 	}
 
-	// Create or open the file used to save device data across restarts.
-	moduleDataDir := os.Getenv("VIAM_MODULE_DATA")
-	filePath := filepath.Join(moduleDataDir, "devicedata.txt")
-	//nolint:gosec
-	file, err := os.OpenFile(filePath, os.O_RDWR|os.O_CREATE, 0o600)
-	if err != nil {
```

Comment:
Where is the creation logic here? Does `os.OpenFile` create a file if it does not exist in the directory?

Reply:
yeah, when given the os.O_CREATE flag os.OpenFile will create a file if it does not exist. 

tbh I should remove the O_CREATE flag, since we now only use this function if we already know the txt file exists